src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +225,182 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
         return minScores;
     }
 
-    private float normalizeSingleScore(final float score, final float minScore, final float maxScore) {
+    private float normalizeSingleScore(final float score, final float minScore, final float maxScore, final LowerBound lowerBound) {
         // edge case when there is only one score and min and max scores are same
         if (Floats.compare(maxScore, minScore) == 0 && Floats.compare(maxScore, score) == 0) {
             return SINGLE_RESULT_SCORE;
         }
-        float normalizedScore = (score - minScore) / (maxScore - minScore);
-        return normalizedScore == 0.0f ? MIN_SCORE : normalizedScore;
+        if (!lowerBound.isEnabled()) {
+            return LowerBound.Mode.IGNORE.normalize(score, minScore, maxScore, lowerBound.getMinScore());
+        }
+        return lowerBound.getMode().normalize(score, minScore, maxScore, lowerBound.getMinScore());
+    }
+
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
+        if (Objects.isNull(params) || !params.containsKey(PARAM_NAME_LOWER_BOUNDS)) {
+            return Optional.empty();
+        }
+
+        List<Pair<LowerBound.Mode, Float>> lowerBounds = new ArrayList<>();
+
+        Object lowerBoundsObj = params.get(PARAM_NAME_LOWER_BOUNDS);
+        if (!(lowerBoundsObj instanceof List<?> lowerBoundsParams)) {
+            throw new IllegalArgumentException("lower_bounds must be a List");
+        }
+
+        if (lowerBoundsParams.size() > MAX_NUMBER_OF_SUB_QUERIES) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
+                    lowerBoundsParams.size(),
+                    MAX_NUMBER_OF_SUB_QUERIES
+                )
+            );
+        }
+
+        for (Object boundObj : lowerBoundsParams) {
+            if (!(boundObj instanceof Map)) {
```

Comment:
```suggestion
            if ((boundObj instanceof Map)==false) {
```
improves readability

Reply:
same as above, I do like the minimalistic syntax

Reply:
@martin-gaievski checking boolean expression with `==` flag is [OpenSearch dev guide lines](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#java-language-formatting-guidelines). We should follow the same

Reply:
ok, let's follow the guideline in core then, although I don't see benefits in that syntax. I want to tweak plugin rules so this is applied automatically, will work on this in a separate issue: https://github.com/opensearch-project/neural-search/issues/1203

Reply:
The above change won't be handled by spotless though. Spotless is used for styling and formatting and the above one is a code change

Reply:
to me this is styling, and this can be done via spotless custom rules

Reply:
I've changed all negations to explicit comparisons with false, should be good for changes in this PR